x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -79,12 +79,12 @@ private static TimeValue delayToNextTime(ClusterName clusterName) {
         return TimeValue.timeValueMillis(next.toInstant().toEpochMilli() - now.toInstant().toEpochMilli());
     }
 
-    public void start() {
+    public synchronized void start() {
```

Comment:
I suppose there is no harm in making this synchronized, it just seems unnecessary as all it does is call `scheduleNext` which is also synchronized.

Reply:
Yes, it's redundant in terms of thread safety, but I thought it was worth making sure the logging order of "Starting ML daily maintenance service" and "Stopping ML daily maintenance service" matched the order that the corresponding operations were performed.  Without this `synchronized` the log could say "Starting" followed by "Stopping", but actually the work of starting was done after the work of stopping.